### DISCUSSION TITLE
egeria & https://openlineage.io/

### DISCUSSION THREAD

**Discussion by jbadeau** [AbsaOSS/spline#959] [Ideas]
Hello,

Do you guys have any plans to integrate with

* [openlineage](https://openlineage.io) a subproject of [egeria](https://github.com/odpi/egeria) ?

This would prob reduce or eliminate the need to maintain agents and spline public model.

It seems to be trying to do what opentelemetry is doing for observability.

**Comment by wajda** [comment]
Never heard about this. Thanks for the info. Will do some research.

**Comment by wajda** [comment]
I made a little research about _Open Lineage_ and I was surprised how conceptually close the OpenLineage API and Spline Producer API models are. The key entities are semantically the same and even the way the model extension points are organized is similar (although the OpenLineage went a step further in their facets which I like).
I think it would definitely make sense to combine efforts of both projects, the question is how. We'll probably start with adopting _OpenLineage_ standard for Spline Agent to Server communication, that will make both parties interoperable with the _OpenLineage_ projects stack.

@jbadeau Do you have any other ideas?

**Comment by ganeshnikumbh** [comment]
@wajda just wanted to check if you have already done something with this topic. I am currently trying few things with openlineage and wanted to know if the spline server can accept and view the lineage events sent by openlineage agent?

**Reply by cerveada** [reply]
Not yet. I'm working on open lineage dispatcher for the spline agent, the server will be done as a second step, so we have both sides covered.
